x/gov/model_helper.go

```diff
@@ -0,0 +1,76 @@
+package gov
+
+import (
+	"bytes"
+	"sort"
+
+	"github.com/iov-one/weave/errors"
+)
+
+// merger is a helper struct to combine Elector sets.
+type merger struct {
+	index map[string]uint32
+}
+
+func newMerger(e []Elector) *merger {
+	r := &merger{
+		index: make(map[string]uint32),
+	}
+	for _, v := range e {
+		r.index[string(v.Address)] = v.Weight
+	}
+	return r
+}
+
+// validate check if the given electors and weights are applicable to the managed elector set.
+func (m merger) validate(diff []Elector) error {
```

Comment:
Why this call?
Why not `merger.merge(diff).validate()`? where `validate() error` takes no args?

Reply:
Validate is called in check without merging. 🤔 I can make this `merger.merge(diff)` to return an `error` and merge both steps. :-)

Reply:
Good idea. Validate without merge was used in check but I have merged both functions into 1 and return an error now.